### PR TITLE
Change `assert!` for `debug_assert!`

### PR DESCRIPTION
**Pull request by josepchetrit12**
I replaced all instances of assert! with debug_assert!, except for the assert! in the test functions.
Issue: [Issue #21 ](https://github.com/lambdaclass/lambdaworks/issues/21)

### CODE REVIEW

**Review by ilitteri — APPROVED** [review]

**Comment by ilitteri** [comment]
There are some conflicts that must be resolved.

**Review by ilitteri — CHANGES_REQUESTED** [review]
The PR looks good! It could be merged when the conflicts are solved.

**Review by jrchatruc — APPROVED** [review]

**Comment by ilitteri** [comment]
Hi! Could you merge from `main` so we can run the CI?

**Comment by josepchetrit12** [comment]
@ilitteri Done!

**Comment by josepchetrit12** [comment]
@ilitteri One question. I passed all the tests except `mul_two_384_bit_integers_works_5`, but I didn't touch anything about it in this PR, should I do something?

**Comment by ilitteri** [comment]
> @ilitteri One question. I passed all the tests except `mul_two_384_bit_integers_works_5`, but I didn't touch anything about it in this PR, should I do something?

That is strange because it is not failing in `main`, I'll take a look

File: math/src/unsigned_integer/element.rs [hunk 1]
```diff
@@ -138,9 +138,8 @@ impl<const NUM_LIMBS: usize> Mul<&UnsignedInteger<NUM_LIMBS>> for &UnsignedInteg
                 t = NUM_LIMBS - 1 - i;
             }
         }
-
-        assert!(
-            n + t < NUM_LIMBS,
+        debug_assert!(
+            n + t + 1 < NUM_LIMBS,
```
Comment (ilitteri) [inline math/src/unsigned_integer/element.rs:142]: Maybe this extra `+ 1` is breaking it?
```suggestion
            n + t < NUM_LIMBS,
```

Comment (josepchetrit12) [reply math/src/unsigned_integer/element.rs]: Yes it was this extra +1, now it passed all the tests!

**Review by MauroToscano — APPROVED** [review]

**Review by ilitteri — APPROVED** [review]

**Pull request merged**